tests/integration/test_storage_iceberg_multistorage/test_table_in_multiple_locations.py

```diff
@@ -0,0 +1,220 @@
+    hadoop_aws_version = "3.3.4"
+    jdk_bundle = "1.12.262"
+
+    builder = (
+        pyspark.sql.SparkSession.builder
+        .appName("IcebergMultiStorageExample")
+        .config("spark.jars.repositories", "https://repo1.maven.org/maven2")
+        .config(
+            "spark.jars.packages",
+            f"org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:{iceberg_version},"
+            f"org.apache.spark:spark-avro_2.12:{spark_version},"
+            f"org.apache.hadoop:hadoop-aws:{hadoop_aws_version},"
+            f"com.amazonaws:aws-java-sdk-bundle:{jdk_bundle}",
+        )
+        .config(
+            "spark.sql.extensions",
+            "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions",
+        )
+        .config(
+            "spark.sql.catalog.spark_catalog",
+            "org.apache.iceberg.spark.SparkSessionCatalog",
+        )
+        .config("spark.sql.catalog.spark_catalog.type", "hadoop")
+        .config(
+            "spark.sql.catalog.spark_catalog.warehouse",
+            "s3a://root/var/lib/clickhouse/user_files/iceberg_data",
+        )
+        .config(
+            "spark.hadoop.fs.s3a.endpoint",
+            f"http://{cluster.minio_ip}:{cluster.minio_port}/",
+        )
+        .config("spark.hadoop.fs.s3a.access.key", minio_access_key)
+        .config("spark.hadoop.fs.s3a.secret.key", minio_secret_key)
+        .config("spark.hadoop.fs.s3a.path.style.access", "true")
+        .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+        .master("local")
+    )
+    return builder.master("local").getOrCreate()
+
+
+@pytest.fixture(scope="package")
+def started_cluster_iceberg():
+    try:
+        cluster = ClickHouseCluster(__file__, with_spark=True)
+        cluster.add_instance(
+            "node1",
+            main_configs=[
+                "configs/config.d/query_log.xml",
+                "configs/config.d/cluster.xml",
+                "configs/config.d/named_collections.xml",
+            ],
+            user_configs=["configs/users.d/users.xml"],
+            with_minio=True,
+            stay_alive=True,
+        )
+
+        logging.info("Starting cluster...")
+        cluster.start()
+
+        prepare_s3_bucket(cluster)
+
+        cluster.spark_session = get_spark(cluster)
+
+        cluster.default_s3_uploader = S3Uploader(cluster.minio_client, cluster.minio_bucket)
+        cluster.default_s3_downloader = S3Downloader(cluster.minio_client, cluster.minio_bucket)
+        cluster.default_local_uploader = LocalUploader(cluster.instances["node1"])
+        cluster.default_local_downloader = LocalDownloader(cluster.instances["node1"])
+
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+def test_table_metadata_in_different_bucket(started_cluster_iceberg):
+    """
+    Test Iceberg table stored in 2 different S3 buckets.
+
+    More complex cases (e.g. metadata on S3, data local) require AVRO manifest file manipulation.
```

Comment:
What kind of manipulations exactly do you mean?

Reply:
Well, we will have to create an iceberg table, then read avro files into something readable, then change necessary paths in them, and then back to avro and put them where they belong onto different storages manually.